### ISSUE TITLE
what is the relationship between xxx.semantic.txt and xxx.semantic.glb?

### ISSUE THREAD

**Issue by zuixiaosanlang** [facebookresearch/habitat-lab#1533]
i get a rgb image from xxx.semantic.glb
![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/17087cb5-fa63-4c03-95e2-531cfdc1db29)
and the floor RGB=(33, 185, 0)

but i cannot find a close color in xxx.semantic.txt about floor.

my code:
![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/17999d1a-3272-474f-97d0-22a93a6ded72)

![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/39075bc1-19c5-439c-8e22-873aa4f20da4)

**Comment by aclegg3** [comment]
Hey @zuixiaosanlang 
I believe your understanding is correct. The hex color in the text file should map to the color of the painted texture region in the .glb asset. We render a flat shaded image of that texture. 

If you can provide the specific scene name here we can take a closer look.

**Comment by jturner65** [comment]
Hi @zuixiaosanlang
The GLAQ4DNUx5U.semantic.txt semantic text file should contain 907 entries, but it seems the file you are reading in only has 23.
[GLAQ4DNUx5U.semantic.txt](https://github.com/facebookresearch/habitat-lab/files/12430743/GLAQ4DNUx5U.semantic.txt)

**Comment by zuixiaosanlang** [comment]
@jturner65 
Hi,
0x87E719 should be rgb : 135, 231, 25.    0x19=1*16+9=25

use MeshLab tool to compare:
![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/bab0b4e8-d47a-4808-8db9-3e8f2f725539)

![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/3b5e1d70-15ec-4e7a-9284-58a99eafaa93)

![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/67cb0395-4102-4a04-8fde-e7949ee4174d)


it seems the floor color in your picture  is more closer to rgb(33, 185, 0)

**Comment by zuixiaosanlang** [comment]
Hi  @aclegg3 
00861-GLAQ4DNUx5U

also you can check other scene, for example 00006-HkseAnWCgqk. it is the same problem. 

If it's my fault, please correct me.
thanks.

**Comment by jturner65** [comment]
> @jturner65 Hi, 0x87E719 should be rgb : 135, 231, 25. 0x19=1*16+9=25
> 
Yes, that was a typo on my part, apologies, and thank you for correcting it.

Please let me know what mechanism you are using to query the color in your original post, so that as I attempt to figure out any potential issues I can follow the same pathway you did.

Thank you!

**Comment by zuixiaosanlang** [comment]
@jturner65 
Hi,
i create tensor:
![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/368b6e66-cdc0-4039-a8ff-e04fdce00637)

get rgb image from xxx.semantic.glb:
![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/271c4fc2-4f2a-4c31-b6a0-d983def17b66)


save rgb image:
![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/d0d289f2-3985-46fa-844d-900108f76f5c)

then i check the floor color in rgb image, compared whith xxx.semantic.txt.

**Comment by jturner65** [comment]
Are you setting up the sensor as a semantic sensor?

**Comment by zuixiaosanlang** [comment]
Hi, @jturner65 
if i set sensor as a semantic sensor, i got all zeros.
![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/a0720f8d-1fb3-4966-b786-f6caa7342cfc)

then next line run failed
batch = batch_obs([observations])

![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/dfdbd6d8-7471-4069-9368-a50364382b0e)

![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/f60429c5-d361-4af3-a118-dea454c17c5f)

**Comment by zuixiaosanlang** [comment]
@jturner65 
i run the code and cannot get the semantic label.

https://aihabitat.org/docs/habitat-sim/image-extractor.html
![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/9652459b-b34f-47a4-9849-21cc0396fff6)

![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/16d8eb9e-5fca-44b1-94b7-0d51f71f315a)

and my env:
![image](https://github.com/facebookresearch/habitat-lab/assets/20105627/08e2521e-4f9c-47cb-bd91-60402fdfc2a5)

thanks.
